src/codegen/compile.ml

```diff
@@ -101,6 +101,15 @@ module Const = struct
     | Float64 of Numerics.Float.t
     | Blob of string
 
+  let lit_eq = function
+    | (Vanilla i, Vanilla j) -> i = j
```

Comment:
Do we have a guarantee that there is no duplication in the vanilla space?

Reply:
Can we get rid of this constructor? Is it used for more than `null`?

Reply:
Phew!